enumerable.rb

```diff
@@ -0,0 +1,200 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    i = 0
+    select_array = []
+    while i < size
+      select_array << yield(self[i])
+      i += 1
+    end
+    select_array
+  end
```

Comment:
- [ ] Eumerables `my_select` returns an array containing all elements of enum for which the given block returns a true value
`[1,2,3,4,5,6].my_select{|num| num > 3}` should return `[4, 5, 6]`